benchexec/runexecutor.py

```diff
@@ -571,11 +557,11 @@ def _setup_cgroup_time_limit(
             return timelimitThread
         return None
 
-    def _setup_cgroup_memory_limit(self, memlimit, cgroups, pid_to_kill):
+    def _setup_cgroup_memory_limit_thread(self, memlimit, cgroups, pid_to_kill):
         """Start memory-limit handler.
         @return None or the memory-limit handler for calling cancel()
         """
-        if memlimit is not None:
+        if memlimit is not None and cgroups.version == 1:
```

Comment:
Is it correct to not use `KillProcessOnOomThread` with cgroups v2? The kernel docs for `memory.max` only claim that it invokes the OOM Killer, but what does this mean precisely?

For example what we do not want is that just some process of the cgroup is killed, we want to terminate the whole run.

If this indeed works as desired, an explaining comment should be added.

Reply:
Previously it did not work as required, but the kernel has a feature that does what we need, so since 6938ef83a8381e34d99859d5a4ba52a40f738afc this is fine.